snp_application/include/snp_application/snp_widget.h

```diff
@@ -3,6 +3,8 @@
 #include <behaviortree_cpp/blackboard.h>
 #include <behaviortree_cpp/loggers/abstract_logger.h>
 #include <QWidget>
+#include <QTextEdit>
+#include <QStackedWidget>
```

Comment:
Let's forward declare these below instead and include the headers in the source file